### PR TITLE
Add REST API for resident IDP/SP properties

### PR DESCRIPTION
**Pull request by emswbandara**
fixes https://github.com/wso2/product-is/issues/8044

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java [hunk 1]
```diff
@@ -52,4 +54,24 @@ public static void setApplicationManagementService(ApplicationManagementService
 
         ConfigsServiceHolder.applicationManagementService = applicationManagementService;
     }
+
+    /**
+     * Get IdentityProviderManager osgi service.
+     *
+     * @return IdentityProviderManager
+     */
+    public static IdentityProviderManager getIdentityProviderManager() {
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java:63]: Remove static... class is defined as a Singleton

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java [hunk 2]
```diff
@@ -95,6 +115,117 @@ public Authenticator getAuthenticator(String authenticatorId) {
...
+        }
+
+        String rememberMePeriod = null;
+        IdentityProviderProperty rememberMeProp = IdentityApplicationManagementUtil.getProperty(
+                residentIdP.getIdpProperties(), IdentityApplicationConstants.REMEMBER_ME_TIME_OUT);
+        if (rememberMeProp != null) {
+            rememberMePeriod = rememberMeProp.getValue();
+        }
+
+        String homeRealmIdStr = residentIdP.getHomeRealmId();
```
Comment (thanujalk) [inline components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:141]: can homeRealmIdStr be null?

Comment (emswbandara) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.v1/src/main/java/org/wso2/carbon/identity/api/server/configs/v1/core/ServerConfigManagementService.java:142]: addressed

File: components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java (hunk 1 above)
Comment (emswbandara) [reply components/org.wso2.carbon.identity.api.server.configs/org.wso2.carbon.identity.api.server.configs.common/src/main/java/org/wso2/carbon/identity/api/server/configs/common/ConfigsServiceHolder.java]: addressed

**Pull request merged**